recipes/icu/build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+cd source
+chmod +x configure install-sh
+
+./configure --prefix="$PREFIX" \
+    --disable-samples \
+    --disable-extras \
+    --disable-layout \
+    --enable-static
```

Comment:
Questions:

1. Do we agree to add `--disable-tests` here? I think @jakirkham said tests only add some test data and don't have to do with actual tests (with `make check` is already doing).
2. Do we really need the static library? I'm -1 for it because it's not needed for Boost and Qt5, and it makes the package bigger.